apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -350,49 +400,60 @@ export async function validateLti13CourseInstance(
+      //
+      // "The Audience SHOULD be the URL of the Authorization Server's Token Endpoint."
+      //
+      // However, `openid-client` changed their behavior in the following commit:
+      // https://github.com/panva/openid-client/commit/0b05217e7f283b75fd93c27c0f8c647f37501a33
+      //
+      // There wasn't a justification provided and this seems to deviate from the OIDC spec.
+      // See a discussion here: https://github.com/panva/openid-client/discussions/730
+      payload.aud = [
+        ...new Set(
+          [lti13_instance.issuer_params.issuer, lti13_instance.issuer_params.token_endpoint].filter(
+            Boolean,
+          ),
+        ),
+      ];
+
+      // FUTURE WORK: Tool SHOULD include the deployment ID as part of the JWT to request a token.
+      // https://www.imsglobal.org/spec/lti/v1p3/#deployment-id
+      // payload['https://purl.imsglobal.org/spec/lti/claim/deployment_id']
+    },
+  };
+
+  const openidClientConfig = await getOpenidClientConfig(lti13_instance, modAssertion);
+
   // Fetch the token
-  const issuer = new Issuer(lti13_instance.issuer_params);
-  const client = new issuer.Client(lti13_instance.client_params, lti13_instance.keystore);
+  tokenSet = await client.clientCredentialsGrant(openidClientConfig, {
+    scope: TOKEN_SCOPES.join(' '),
+  });
 
-  tokenSet = await client.grant(
-    {
-      grant_type: 'client_credentials',
-      scope: TOKEN_SCOPES.join(' '),
-    },
-    {
-      clientAssertionPayload: {
-        // Canvas requires the `aud` claim the be (or contain) the token endpoint:
-        // https://github.com/instructure/canvas-lms/blob/995169713440bd8305854d440b336911a734c38f/lib/canvas/oauth/client_credentials_provider.rb#L26-L29
-        // https://github.com/instructure/canvas-lms/blob/995169713440bd8305854d440b336911a734c38f/lib/canvas/oauth/asymmetric_client_credentials_provider.rb#L24
-        //
-        // From what we can tell, the OIDC spec requires this as well, see "private key jwt"
-        // here: https://openid.net/specs/openid-connect-core-1_0.html#rfc.section.9
-        //
-        // "The Audience SHOULD be the URL of the Authorization Server's Token Endpoint."
-        //
-        // However, `openid-client` changed their behavior in the following commit:
-        // https://github.com/panva/openid-client/commit/0b05217e7f283b75fd93c27c0f8c647f37501a33
-        //
-        // There wasn't a justification provided and this seems to deviate from the OIDC spec.
-        // See a discussion here: https://github.com/panva/openid-client/discussions/730
-        aud: [...new Set([issuer.issuer, issuer.token_endpoint].filter(Boolean))],
-      },
-    },
-  );
+  debug('getAccessToken');
+  debug(tokenSet);
 
   // Store the token for reuse
-  const expires_at = tokenSet.expires_at ? tokenSet.expires_at * 1000 : Date.now();
+  const expires_at = tokenSet.expires_in ? Date.now() + tokenSet.expires_in * 1000 : Date.now();
```

Comment:
I left the database column as `access_tokenset_expires_at` because it's a calculated timestamp (rather than a duration in seconds, like `expires_in`). I'm not sure why the previous library called it `expires_at` instead of the OIDC standard `expires_in`.